### ISSUE TITLE
Ransack 4.0 allowlist be an optional config

### ISSUE THREAD

**Issue by gvkhna** [activerecord-hackery/ransack#1482]
This allowlist enforcement actually needs to be able to be turned off. I understand the security protections trying to be put in place by this change. But for many apps it's not even needed, internal only apps that aren't even internet accessible don't need additional any additional attention. I can understand default requiring this but a simple config of some kind to turn it off would be needed as well.

TLDR: This was a breaking change and instituted quite abruptly, and so everyone probably is just pinning to the older version.

**Comment by sk-** [comment]
Another reason to offer an opt out mechanism is the performance degradation reported in #1462

**Comment by Tmpv** [comment]
The old behaviour is quite simple to implement and is actually [mentioned](https://github.com/activerecord-hackery/ransack/pull/1400/files#diff-18b0303eda26dc0eb1c3cd5912aa46759501b06e843dedc53c627925a2aa5a5fR31-R46) in one of the tests implementing the new requirement.

But this is not mentioned in the official documentation.

**Comment by gvkhna** [comment]
@Tmpv Am i missing something, I added that to ApplicationRecord and I get undefined method `authorizable_ransackable_attributes` on a class that descends from ApplicationRecord?

**Comment by MmKolodziej** [comment]
There's also so many other access control mechanisms that work on a different layer than models that developers might choose to utilize. Strong params, explicitly defined filters in ActiveAdmin (and I promise that so many application will only use Ransack as a dependency of ActiveAdmin), form objects with params validation, etc.

While giving the option to perform the validation inside models is great, there really should be a simple way to opt out of it.

**Comment by MmKolodziej** [comment]
Furthermore, if there are gems which define AR models (e.g. mobility defines them dynamically), they might inherit directly from `ActiveRecord::Base`, leaving no good place to define the necessary methods.

**Comment by budu** [comment]
Also support for polymorphic associations has been completely broken:

https://activerecord-hackery.github.io/ransack/going-further/polymorphic-search/

**Comment by scarroll32** [comment]
> [!NOTE]
> Comment from **Claude** (Claude Code), acting on behalf of @scarroll32.

The opt-out exists, it is just not documented yet: in `ApplicationRecord` define `def self.ransackable_attributes(_ = nil) = authorizable_ransackable_attributes` and `def self.ransackable_associations(_ = nil) = authorizable_ransackable_associations`, which restores the pre-4.0 everything-searchable behaviour for every model that inherits from it. For models that inherit from `ActiveRecord::Base` directly (gems like mobility), put the same two methods in a module and `ActiveSupport.on_load(:active_record) { extend ThatModule }` from an initializer. A global config flag would be this in a different place, so one is not being added; the recipe is going into the docs. For questions and design discussions please use the [Discussions forum](https://github.com/activerecord-hackery/ransack/discussions); the issue tracker is kept for reproducible bugs.
